packages/next/cli/next-lint.ts

```diff
@@ -131,7 +131,9 @@ const nextLint: cliCommand = (argv) => {
     printAndExit(`> No such directory exists as the project root: ${baseDir}`)
   }
 
-  const dirs: string[] = args['--dir']
+  const conf = await loadConfig(PHASE_PRODUCTION_BUILD, baseDir)
+
+  const dirs: string[] = args['--dir'] ?? conf.eslint?.dirs
```

Comment:
Could you also update the [ESLint docs](https://github.com/vercel/next.js/blob/2a5db0b58c9416381568b9716861b598b17a7015/docs/basic-features/eslint.md#linting-custom-directories) to remove the mention that this only works for production builds? It's probably still worth keeping the `--dir` flag, so could we also include a line on precedence? (i.e. Using `--dir` will take precedence over `conf.eslint?.dirs`)